src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,182 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
+    """
+    # Always upload the manifest file to case root /Manifest with the original file name.
+    manifest_path = manifest_path + "/Manifests/" + Path(manifest_file).name
+
+    # S3 uploader.
+    upload = S3AssetUploader(session=boto_session)
+    with open(manifest_file) as manifest:
+        upload.upload_bytes_to_s3(
+            bytes=BytesIO(manifest.read().encode("utf-8")),
+            bucket=s3_bucket_name,
+            key=manifest_path,
+            progress_handler=logger.echo,
+            extra_args=s3_metadata,
+        )
+
+
+def _manifest_download(
+    download_dir: str,
+    farm_id: str,
+    queue_id: str,
+    job_id: str,
+    boto3_session: boto3.Session,
+    step_id: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDownloadResponse:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to download the Job Attachment manifest for a Job, and optionally dependencies for Step.
+    download_dir: Download directory.
+    farm_id: The Deadline Farm to download from.
+    queue_id: The Deadline Queue to download from.
+    job_id: Job Id to download.
+    boto_session: Boto3 session.
+    step_id: Optional[str]: Optional, download manifest for a step
+    logger: Click Logger instance to print to CLI as test or JSON.
+    return ManifestDownloadResponse Downloaded Manifest data. Contains source S3 key and local download path.
+    """
+
+    # Deadline Client and get the Queue to download.
+    deadline = boto3_session.client("deadline", config=get_default_client_config())
+
+    queue: dict = deadline.get_queue(
+        farmId=farm_id,
+        queueId=queue_id,
+    )
+
+    # assume queue role - session permissions
+    queue_role_session: boto3.Session = _get_queue_user_boto3_session(
+        deadline=deadline,
+        base_session=boto3_session,
+        farm_id=farm_id,
+        queue_id=queue_id,
+        queue_display_name=queue["displayName"],
+    )
+
+    # Get S3 prefix
+    s3_prefix: Path = Path(queue["jobAttachmentSettings"]["rootPrefix"], S3_MANIFEST_FOLDER_NAME)
+
+    # Queue's Job Attachment settings.
+    queue_s3_settings = JobAttachmentS3Settings(**queue["jobAttachmentSettings"])
+
+    # Capture a list of success and failed to download files for JSON output.
+    successful_downloads: List[ManifestDownload] = []
+    failed_downloads: List[str] = []
+
+    # Utility function to build up manifests by root.
+    manifests_by_root: Dict[str, List[BaseAssetManifest]] = dict()
+
+    def add_manifest_by_root(
+        manifests_by_root: Dict[str, list], root: str, manifest: BaseAssetManifest
+    ):
+        if root not in manifests_by_root:
+            manifests_by_root[root] = []
+        manifests_by_root[root].append(manifest)
+
+    # Get input_manifest_paths from Deadline GetJob API
+    job: dict = deadline.get_job(farmId=farm_id, queueId=queue_id, jobId=job_id)
```

Comment:
`get_job` returns `Job` type. Data unboxing could directly use the job data class definition rather than dict key access.

Reply:
Yes, but have you seen how much code `get_job` runs! 

https://github.com/aws-deadline/deadline-cloud/blob/61fb59581cd4c5d548077612721f7ee1b7527049/src/deadline/job_attachments/_aws/deadline.py#L67-L108

Reply:
What do you mean, I thought you've already calling this function?

Reply:
Code reuse brings benefits in terms of efficiency, consistency, and maintainability. The additional code is boxing the data and error handling, which can be useful here. For example, `attachment` is not required in GetJobResponse, L362 could lead to KeyError, same goes for `inputManifestPath` in L364.

Reply:
But isn't that case regardless? All the code has done is add an error message instead of where it actually failed. Overtime, as get_job changes, it runs more and more code that may or may not have tangible benefits. 

If it was a simple typed dict class I would be more interested. No better or worse solution here.

Reply:
Moved the [] accessors to .get() with default handling as discussed offline.